src/main/native/JitsiCipherGcmAuthOnlyOpenSsl.c

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright @ 2016 - present 8x8, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "org_jitsi_srtp_crypto_OpenSslAesGcmAuthOnlyCipherSpi.h"
+
+#include <openssl/modes.h>
+#include <openssl/evp.h>
+#include <openssl/crypto.h>
+
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <assert.h>
+
+typedef struct {
+    EVP_CIPHER_CTX* cipher;
+    GCM128_CONTEXT* gcm;
+} AES_GCM_CONTEXT;
+
+
+/* Encrypt one block with an EVP_CIPHER_CTX, with a function signature
+ * compatible with block128_f */
+static void EVP_encrypt(const unsigned char in[16],
+    unsigned char out[16], const void *ctx)
+{
+    EVP_CIPHER_CTX* ctx_ = (EVP_CIPHER_CTX*)ctx;
+    int len = 16;
+    EVP_EncryptUpdate(ctx_, out, &len, in, len);
+    assert(len == 16);
+}
+
+/*
+ * Class:     org_jitsi_srtp_crypto_OpenSslAesGcmAuthOnlyCipherSpi
+ * Method:    CRYPTO_gcm128_new
+ * Signature: ()J
+ */
+JNIEXPORT jlong JNICALL Java_org_jitsi_srtp_crypto_OpenSslAesGcmAuthOnlyCipherSpi_CRYPTO_1gcm128_1new
+  (JNIEnv *env, jclass clazz)
+{
+    AES_GCM_CONTEXT* ctx = malloc(sizeof(AES_GCM_CONTEXT));
+    if (ctx == NULL)
+    {
+        goto fail;
+    }
+    OPENSSL_cleanse(ctx, sizeof(*ctx));
```

Comment:
No, not really.

The only subtle difference is that there are cases where compilers can optimize out a `memset` immediately followed by `free`, since they can tell the results of the `memset` aren't used, whereas I believe `OpenSSL_cleanse` is supposed to do the appropriate compiler magic to avoid that.  But I don't think that's relevant for this code path.

Nonetheless, I've generally considered it good code style to use the appropriate cleanse function as needed for cryptographic APIs.

Reply:
Okay, I was just asking because then it is redundant with lines 62 and 68, unless of course you expect the AES_GCM_CONTEXT structure to grow.

Reply:
Yes, I think this was an artifact of an earlier approach I was taking to this code, where I used a different AES API.  I'll remove it.